src/libstore/filetransfer.cc

```diff
@@ -216,6 +216,19 @@ struct curlFileTransfer : public FileTransfer
                         encoding = trim(line.substr(i + 1));
                     else if (name == "accept-ranges" && toLower(trim(line.substr(i + 1))) == "bytes")
                         acceptRanges = true;
+                    else if (name == "x-amz-meta-nix-is-immutable" || name == "x-nix-is-immutable") {
```

Comment:
I wonder: would it be better to check something like `starts_with("x-") && ends_with("nix-is-immutable")` so that any other place that offers similar functionality to Amazon won't need to be hardcoded in here?

Reply:
Pretty much everyone with similar functionality (where you can't set an arbitrary header) has adopted x-amz-meta- after s3 took over the world. I like that it is an explicit set of headers to be looked for, but no real strong opinion.

Reply:
It seems a bit scary to have a wildcard match in header names. I think it's better to be strict in what headers we accept.